mlir/include/mlir/Dialect/Math/Transforms/Passes.h

```diff
@@ -48,6 +48,27 @@ struct MathPolynomialApproximationOptions {
 void populatePolynomialApproximateTanhPattern(RewritePatternSet &patterns);
 void populatePolynomialApproximateErfPattern(RewritePatternSet &patterns);
 
+// Adds patterns to convert to f32 around math functions for which `predicate`
+// returns true.
+void populateMathF32ExpansionPatterns(
+    RewritePatternSet &patterns,
+    const std::function<bool(StringRef)> &predicate);
```

Comment:
Use function_ref -- it's guaranteed not to allocate

Reply:
maybe, but I have found it to be a bit too much of a footgun. Id just maybe stick with std::function copy by value.